File: 0041-first-missing-positive/0041-first-missing-positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        poss = [i for i in nums if i > 0]
        if not poss:
            return 1
        m = min(poss)
        M = max(poss)
        if m > 1:
            return 1
        posset = set(poss)
        for i in range(m, M+2):
            if i in posset:
                continue
            return i

File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        Set<Integer> numset = Arrays.stream(nums)
                .filter(it -> it > 0)
                .boxed()
                .collect(Collectors.toSet());
        if (numset.isEmpty() || numset.stream().mapToInt(Integer::valueOf).min().getAsInt() > 1) {
            return 1;
        }
        for (int i = 1; ; i++) {
            if (!numset.contains(i))
                return i;
        }
    }
}